shelf/bucket_update/search_updater.py

```diff
@@ -57,9 +58,46 @@ def run(self):
             self.update_manager.bulk_update(bulk_update)
             all_id_list = all_id_list + bulk_update.keys()
 
+        self.remove_unlisted_documents_per_bucket(all_id_list)
+
+    def prune(self):
```

Comment:
I don't think it needs to be documented. Its general coding practice to make code modular and there are not a lot of good guidelines for this that I have heard.

I've heard "code should only do a single thing" but "a single thing" means a lot of different things to people. I've heard arbitrary lines of code before breaking it out. Thats better but it often doesn't hold true.